app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def login
+    id = params.permit(:login_id)[:login_id].to_i
```

Comment:
`params.permit(:login_id)[:login_id]` is probably overkill here. You typically use `permit` to protect yourself when you're passing in a bunch of stuff to something that will modify your database. Since you're asking params for `:login_id` directly, `permit` isn't doing much for you.